IdentityServer/Quickstart/Account/ExternalController.cs

```diff
@@ -89,14 +98,138 @@ public IActionResult Challenge(string provider, string returnUrl)
                                                  }
                                              };
 
+            HttpContext.Response.Cookies.Append("returnUrl", returnUrl);
+
             return Challenge(props, provider);
         }
 
+        /// <summary>
+        /// The callback endpoint for the fontys single sign on.
+        /// </summary>
+        /// <param name="code">The code.</param>
+        /// <param name="state">The state.</param>
+        /// <param name="session_state">State of the session.</param>
```

Comment:
What code or state?
Use more descriptive names for what these parameters are used for.